services/storehost/replicationJobRunner.go

```diff
@@ -224,9 +234,35 @@ func (runner *replicationJobRunner) run() {
 		}).Info(`replication for extent started`)
 	}
 
+	var maxConsecutiveFailures int
+	for existingFailedJob, failedTimes := range runner.failedJobs {
+		if _, ok := currentFailedJobs[existingFailedJob]; !ok {
+			delete(runner.failedJobs, existingFailedJob)
+		} else {
+			newFailedTimes := failedTimes + 1
+			runner.failedJobs[existingFailedJob] = newFailedTimes
+			if newFailedTimes > maxConsecutiveFailures {
+				maxConsecutiveFailures = newFailedTimes
+			}
+		}
+	}
+	for currentFailedJob, _ := range currentFailedJobs {
+		if failedTimes, ok := runner.failedJobs[currentFailedJob]; !ok {
+			runner.failedJobs[currentFailedJob] = 1
+		} else {
+			runner.logger.WithFields(bark.Fields{
+				common.TagExt:  common.FmtExt(currentFailedJob),
+				`failed times`: failedTimes,
+			}).Info(`replication job failed for at least twice`)
```

Comment:
it looks like this case is the same as the one in the else block at line 242 .. basically handling the intersection between runner.failedJobs and currentFailedJobs. You could probably move this log up there. Btw, I am wondering if you really need this log anyway .. since you are already logging when the error occurs. But I'll leave it to you.